src/ast/passes/semantic_analyser.cpp

```diff
@@ -3553,60 +3601,45 @@ SizedType SemanticAnalyser::create_key_type(const SizedType &expr_type,
   return new_key_type;
 }
 
-void SemanticAnalyser::update_key_type(const Map &map,
-                                       const SizedType &new_key_type)
+void SemanticAnalyser::update_key_type(SizedType &current_key_type,
```

Comment:
This method deserves a comment. IIUC, originally, it not only updated `current_key_type` but also inserted entries into `map_key_` (which is a bit confusing tbh). Documenting the purpose of the method would make it easier to understand when it should be used.

Reply:
I split this out into two separate methods to make it more understandable, which resulted in some reduced code.